core/src/main/java/org/apache/calcite/rel/metadata/RelMdUniqueKeys.java

```diff
@@ -131,7 +159,7 @@ public Set<ImmutableBitSet> getUniqueKeys(Project rel, RelMetadataQuery mq,
         Util.transform(program.getProjectList(), program::expandLocalRef));
   }
 
-  private static Set<ImmutableBitSet> getProjectUniqueKeys(SingleRel rel, RelMetadataQuery mq,
+  private Set<ImmutableBitSet> getProjectUniqueKeys(SingleRel rel, RelMetadataQuery mq,
```

Comment:
For most use-cases we shouldn't hit the limit so I would say that it is not that important. Moreover, it would be difficult to provide this information via the regular result of the method. An alternative, that crossed my mind is throwing an well defined exception when the limit is reached instead of truncating but this may be a less desired behavior.